flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -454,6 +510,15 @@ private void initializeState(BufferedRequestState<RequestEntryT> state) {
     @Override
     public void close() {}
 
+    private int getNextBatchSizeLimit() {
+        return Math.min(maxBatchSize, inflightMessagesLimit);
+    }
+
+    @VisibleForTesting
+    int getInflightMessagesLimit() {
+        return inflightMessagesLimit;
+    }
```

Comment:
Instead of adding this, can you reuse the `AIMDRateLimitingStrategy.java` class from previous PR and make it `@Internal`? This would retain the existing interface and remove this code smell